### ISSUE TITLE
Unable to link to overloaded functions/priocedures

### ISSUE THREAD

**Issue by MrUrdam** [NaturalDocs/NaturalDocs#70]
I tried the following to link to a overloaded function but it does not work.
```SQL
/*
 * Procedure: proc_name
 * description
 *
 * Parameters:
 * p_a - p_a description.
 * p_b - p_b description.
 */
MEMBER PROCEDURE proc_name(
  p_a CHAR,
  p_b NUMBER
),

/*
 * Procedure: proc_name
 * description
 *
 * See Also:
 * Procedure <proc_name(NUMBER, VARCHAR2, BOOLEAN)> has been deprecated. <-- This links to the above proc_name.
 *
 * Parameters:
 * p_a - p_a description.
 * p_b - p_b description.
 * p_c - p_c description.
 */
MEMBER PROCEDURE proc_name(
  p_a NUMBER,
  p_b VARCHAR2,
  p_c BOOLEAN
),

**Comment by NaturalDocs** [comment]
This is a bug I've known about for a bit.  Natural Docs knows which one it should be linked to but the generated HTML anchors are the same so it always goes to the top one if there's more than one in the same file.  I have to figure out a good way to handle it.  This will be fixed sooner rather than later but I can't give you a timeline.

**Comment by NaturalDocs** [comment]
Try this version: https://www.naturaldocs.org/download/temp/Natural_Docs_temp8.zip

**Comment by MrUrdam** [comment]
RC1 works. Thank you :)
Another behavior I found during testing this build:

- I press a link to a overloaded function
- I jump to the correct location on the page
- I scroll down or up manually
- I click the link to the exact same function again
- The jump does not work, because the browser is still on the same URL.

**Comment by MrUrdam** [comment]
Maybe you could "reset" the location in the event, after you moved/scrolled to the desired location.
This would fire the event again, but could be handled/ignored in it.
Not a clean solution but maybe its possible. 
But I do not know if it dies break other stuff. 

Because it is just a minor issue, a fix is not that important for myself.

**Comment by NaturalDocs** [comment]
That's the problem, there's no event.  My code receives no notification from the browser that any attempted navigation occurred.

However, I did find a partial fix.  I can tie into the page-wide onmousedown handler and check if a link was clicked that goes to the same location as the current one, and then manually call the onhashchange handler from there.  This makes the summary pane always work, so you can click on a function there, scroll away, and then click on it again and it will work.

This doesn't work with links in the content pane though.  If you have a link to FunctionA in the description of FunctionB, click it, and then scroll down to FunctionB to click it again, it won't work.  I can't use the same trick because an iframe can't access its parent's URL.  However, this scenario should be far less common than using the summary pane so I consider it a win.

The fix will be included in 2.2, or if you want it now you can apply this small patch to the JavaScript yourself:
https://github.com/NaturalDocs/NaturalDocs/commit/1bbd5823dcf47e2ecc6ef26bff66189f122552ac

**Comment by MrUrdam** [comment]
Thank you for the fix :)

**Comment by NaturalDocs** [comment]
[Fixes for both issues are in Natural Docs 2.2.](https://www.naturaldocs.org/download/)
